open-telemetry/opentelemetry-js
# fix(instrumentation-fetch, instrumentation-xml-http-request) content length attributes now prese

## Which problem is this PR solving?

`http.response_content_length` and `http.response_content_length_uncompressed` attributes are not present on network spans from `instrumentation-xml-http-request` and `instrumentation-fetch` plugins when the `ignoreNetworkEvents` configuration property is set to `true`

This PR changes `addSpanNetworkEvents` to include the options to skip over adding network events but still include the previously mentioned properties.

Fixes #5229 

## Short description of the changes

By skipping over the call to `addSpanNetworkEvents` the mentioned attributes were not getting added to the main and preflight network spans. We move the logic from the calls to the method into the `addSpanNetworkEvents` call itself with a third boolean parameter. This will disable adding network events spans while still adding the attributes. 

While fixing this we considered several different changes, we went with adding an optional paramiter to `addSpanNetworkEvents`

## Type of change

Please delete options that are not relevant.


## How Has This Been Tested?

Automated unit tests were written as well as testing with the fetch and xhr examples to verify the attributes are being added with the proposed code.

 1. Open the examples/opentelemetry-web/examples/fetchexample
 2. Start example
 3. Note that the GET event has a http.response_content_length attribute
 4. Add the ignoreNetworkEvents: true config to the FetchInstrumentation in the index.js file
 5. Run example
 6. The http.response_content_length attribute should still be present on the GET span

## Checklist:
